jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/db2/DB2SqlGenerationVisitor.java

```diff
@@ -247,91 +221,145 @@ public String visit(final SqlFunctionScalar function) throws AdapterException {
+            break;
         case BIT_TO_NUM:
-                    sql = sql.replaceFirst("^BIT_TO_NUM", "BIN_TO_NUM");
-                    break;
-        case NULLIFZERO: {
-                    final List<String> argumentsSql = new ArrayList<>();
-                    for (final SqlNode node : function.getArguments()) {
-                        argumentsSql.add(node.accept(this));
-                    }
-                    final StringBuilder builder = new StringBuilder();
-                    builder.append("NULLIF(");
-                    builder.append(argumentsSql.get(0));
-                    builder.append(", 0)");
-                    sql = builder.toString();
-                    break;
-                }
-        case ZEROIFNULL: {
-                    final List<String> argumentsSql = new ArrayList<>();
-                    for (final SqlNode node : function.getArguments()) {
-                        argumentsSql.add(node.accept(this));
-                    }
-                    final StringBuilder builder = new StringBuilder();
-                    builder.append("IFNULL(");
-                    builder.append(argumentsSql.get(0));
-                    builder.append(", 0)");
-                    sql = builder.toString();
-                    break;
-                }
-        case DIV: {
-                    final List<String> argumentsSql = new ArrayList<>();
-                    for (final SqlNode node : function.getArguments()) {
-                        argumentsSql.add(node.accept(this));
-                    }
-                    final StringBuilder builder = new StringBuilder();
-                    builder.append("CAST(FLOOR(");
-                    builder.append(argumentsSql.get(0));
-                    builder.append(" / FLOOR(");
-                    builder.append(argumentsSql.get(1));
-                    builder.append(")) AS DECIMAL(36, 0))");
-                    sql = builder.toString();
-                    break;
-                }
+            sql = sql.replaceFirst("^BIT_TO_NUM", "BIN_TO_NUM");
+            break;
+        case NULLIFZERO:
+            sql = getNullZero(function, "NULLIF(");
+            break;
+        case ZEROIFNULL:
+            sql = getNullZero(function, "IFNULL(");
+            break;
+        case DIV:
+            sql = getDiv(function);
+            break;
         default:
             break;
         }
+        return sql;
+    }
 
-        final boolean isDirectlyInSelectList = (function.hasParent() && function.getParent().getType() == SqlNodeType.SELECT_LIST);
-        if (isDirectlyInSelectList && scalarFunctionsCast.contains(function.getFunction())) {
-            // Cast to FLOAT because result set metadata has precision = 0, scale = 0
-            sql = "CAST("  + sql + " AS FLOAT)";
+    private String getDiv(final SqlFunctionScalar function) throws AdapterException {
+        final List<String> argumentsSql = new ArrayList<>();
```

Comment:
Put `function.getArguments()` into a local variable. Initialize `ArrayList` with the size of the argument list.

Repeat for all similar situations.

Reply:
done